app/src/main/java/com/github/polypoly/app/base/game/bonus_card/BonusCard.kt

```diff
@@ -0,0 +1,35 @@
+package com.github.polypoly.app.base.game.bonus_card
+
+import com.github.polypoly.app.base.game.Player
+
+/**
+ * A class that represent a bonus card
+ * @property id The id of the [BonusCard]
+ * @property title The title of the [BonusCard]
+ * @property description The description of the [BonusCard]
+ */
+enum class BonusCard(
+    val id: Long,
+    val title: String,
+    val description: String,
+) {
+
+    TEACHING_ASSISTANT_PAYDAY(0, title = "Teaching assistant payday", description = "You finally got your student assistant salary you've been waiting for after spending hours helping BA1s debug their code. Collect 100CHF!"),
+    TUITION_FEES_TO_PAY(1, title = "Tuition fees to pay!", description = "It's time to pay your tuition fees: you have passed the is academia deadline for far too long... Lose 100CHF.");
+
+    /**
+     * Apply the bonus of the [BonusCard] to the [Player]
+     * @param player the [Player] to apply the bonus to
+     */
+    fun applyBonus(player: Player) {
+        when(this) {
```

Comment:
it might make more sense for each bonus card to contain its own action, rather than split the definition of each card and the definition of which action each card does? As it stands this file mixes functional and object-oriented programming in  a bit of an odd way

Reply:
Yes, it clearly makes sense. I'm going to fix this